src/components/Tabs/Tabs.tsx

```diff
@@ -113,21 +114,50 @@ export const TabBase = ({
 
     const valueIndex = nameToIndexMap[value];
 
-    const animationRef = useRef(new Animated.Value(0));
+    const positionAnimationRef = useRef(new Animated.Value(0));
+    const widthAnimationRef = useRef(new Animated.Value(0));
     const scrollViewRef = useRef<RNScrollView>(null);
     const scrollViewPosition = useRef(0);
 
-    const tabItemPositions = useRef<
-        Array<{ position: number; width: number; contentWidth: number }>
-    >([]);
+    const tabItemPositions = useRef<Array<{ width: number; contentWidth: number }>>([]);
     const [tabContainerWidth, setTabContainerWidth] = useState(0);
 
+    const itemPositionsMap = useMemo(() => {
+        return tabItemPositions.current.reduce((acc, item, index) => {
+            const previousItemsWidth = tabItemPositions.current.reduce((totalWidth, _item, i) => {
```

Comment:
`tabItemPositions.current.slice(0, index).reduce((x, y) => x+y)`

this doesn't loop over items you do not want to loop over.
minimal change.